src/transformers/models/mpt/modeling_mpt.py

```diff
@@ -567,9 +569,11 @@ def forward(
             shift_labels = labels[..., 1:].contiguous()
             batch_size, seq_length, vocab_size = shift_logits.shape
             # Flatten the tokens
-            loss_fct = CrossEntropyLoss()
-            loss = loss_fct(
-                shift_logits.view(batch_size * seq_length, vocab_size), shift_labels.view(batch_size * seq_length)
+            loss = self.loss_function(
+                shift_logits.view(batch_size * seq_length, vocab_size),
+                shift_labels.view(batch_size * seq_length),
+                vocab_size=vocab_size,
+                **kwargs,
             )
```

Comment:
bit weird, the refactor here should make you only have to pass inputs and the shifts will hapen instide